src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1672,7 +1672,7 @@ protected override void TransformManagedToNative(ILCodeStream codeStream)
         {
             ILEmitter emitter = _ilCodeStreams.Emitter;
 
-            if (In && !Out && !IsManagedByRef)
+            if (In && !Out && !IsManagedByRef && MarshallerType == MarshallerType.Argument)
```

Comment:
I would expect this condition to be identical to:

https://github.com/dotnet/runtime/blob/170635417c3a87afd21c6138bdbdf2fd2874148b/src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.cs#L1476-L1480

Reply:
Maybe both of these places should be changed to:
```
MarshalDirection == MarshalDirection.Forward 
     && MarshallerType == MarshallerType.Argument 
     && !IsManagedByRef 
     && In 
     && !Out
```
?